hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -489,6 +491,15 @@ public class HMaster extends HBaseServerBase<MasterRpcServices> implements Maste
   public static final String WARMUP_BEFORE_MOVE = "hbase.master.warmup.before.move";
   private static final boolean DEFAULT_WARMUP_BEFORE_MOVE = true;
 
+  /**
+   * Use RSProcedureDispatcher instance to initiate master -> rs remote procedure execution. Use
+   * this config to provide customized RSProcedureDispatcher.
```

Comment:
What are we planning here @virajjasani ?

Reply:
With the current master's proc executor initialization, its becoming very clumsy to provide a setter method for test to override using custom logic for RSProcedureDispatcher.

Introduced config here to achieve the same but the plan is only for test to use new extended implementation of RSProcedureDispatcher.